Earthfile

```diff
@@ -339,181 +356,44 @@ base-image:
-        # compress firmware
-        RUN find /usr/lib/firmware -type f -execdir zstd --rm -9 {} \+
-        # compress modules
-        RUN find /usr/lib/modules -type f -name "*.ko" -execdir zstd --rm -9 {} \+
-    END
-
-    IF [ "$BUILD_INITRD" = "true" ]
-      IF [ "$FLAVOR" = "debian" ]
-        RUN rm -rf /boot/initrd.img-*
-      END
-
-      RUN --no-cache kernel=$(ls /lib/modules | head -n1) && depmod -a "${kernel}"
-
-      IF [ -f "/usr/bin/dracut" ]
-          # Regenerate initrd if necessary
-          RUN --no-cache kernel=$(ls /lib/modules | head -n1) && dracut -f "/boot/initrd-${kernel}" "${kernel}" && ln -sf "initrd-${kernel}" /boot/initrd
-      END
-
-      IF [ -f "/sbin/mkinitfs" ]
-        # Proper config files with immucore and custom initrd should already be in there installed by framework
-        RUN --no-cache kernel=$(ls /lib/modules | head -n1) && mkinitfs -o /boot/initrd $kernel
-      END
-    END
-
-    # Set /boot/vmlinuz pointing to our kernel so kairos-agent can use it
-    # https://github.com/kairos-io/kairos-agent/blob/0288fb111bc568a1bfca59cb09f39302220475b6/pkg/elemental/elemental.go#L548   q
-
-    # this is generally present on rhel based systems, but it doesn't hurt to remove in any case
-    RUN rm -rf /boot/initramfs-* || true
-
-    IF [ ! -e "/boot/vmlinuz" ]
-        IF [ -e "/boot/vmlinuz-lts" ]
-            # Alpine provides the kernel under this name
-            RUN ln -sf /boot/vmlinuz-lts /boot/vmlinuz
-        END
-        IF [ -e "/boot/vmlinuz-rpi4" ]
-            # Alpine-rpi provides the kernel under this name
-            RUN ln -sf /boot/vmlinuz-rpi4 /boot/vmlinuz
-        END
-        # If it's an ARM flavor, we want a symlink here from zImage/Image
-        # Check that its not a symlink already or grub will fail!
-        IF [ -e "/boot/Image" ] && [ ! -L "/boot/Image" ]
-            RUN ln -sf Image /boot/vmlinuz
-        ELSE IF [ -e "/boot/zImage" ]
-            IF  [ ! -L "/boot/zImage" ]
-                RUN ln -sf zImage /boot/vmlinuz
-            ELSE
-                RUN kernel=$(ls /boot/zImage-* | head -n1) && if [ -e "$kernel" ]; then ln -sf "${kernel#/boot/}" /boot/vmlinuz; fi
-            END
-        ELSE
-            # Debian has vmlinuz-VERSION
-            RUN kernel=$(ls /boot/vmlinuz-* | head -n1) && if [ -e "$kernel" ]; then ln -sf "${kernel#/boot/}" /boot/vmlinuz; fi
-            RUN kernel=$(ls /boot/Image-* | head -n1) && if [ -e "$kernel" ]; then ln -sf "${kernel#/boot/}" /boot/vmlinuz; fi
-        END
-    END
-
-    RUN rm -rf /tmp/*
-
-    COPY ./naming.sh .
-    RUN ./naming.sh container_artifact_name > ./IMAGE
-    RUN rm naming.sh
-
-    # luet cleanup
-    RUN luet cleanup
-    RUN rm -rf /var/luet
+      --build-arg VARIANT=$VARIANT \
+      --build-arg VERSION=$KAIROS_VERSION \
+      --build-arg K3S_VERSION=$K3S_VERSION \
+      --build-arg FRAMEWORK_VERSION=master \
+      -f +kairos-dockerfile/Dockerfile \
```

Comment:
wait... you can actually do that? Where does that file live?

Reply:
yeah, this is quite cool, it lives only in memory so it doesn't pollute the local env